shapes/produce_shapes.py

```diff
@@ -0,0 +1,1173 @@
+        smChannelsDict["mt"].cuts.remove("muon_iso")
+        smChannelsDict["mt"].cuts.add(
+            Cut("(iso_1<0.5)*(iso_1>=0.15)", "muon_iso_loose"))
+        smChannelsDict["et"].cuts.remove("ele_iso")
+        smChannelsDict["et"].cuts.add(
+            Cut("(iso_1<0.5)*(iso_1>=0.1)", "ele_iso_loose"))
+        smChannelsDict["tt"].cuts.get("os").invert()
+        smChannelsDict["em"].cuts.get("os").invert()
+
+    selectedChannels = set(args.channels) - {None}
+    selectedCategories = set(args.categories)
+
+    selectedChannelsTuples = {
+        c_: smChannelsDict[c_] for c_ in selectedChannels}.items()
+    selectedChannelsTuplesNoEM = {
+        c_: smChannelsDict[c_] for c_ in selectedChannels if c_ != "em"}.items()
+
+    pnameToEstD = {
+        "data_obs": DataEstimation,
+        "EMB": ZTTEmbeddedEstimation,
+        "TTT": TTTEstimation,
+        "VVT": VVTEstimation,
+        "ZTT": ZTTEstimation,
+        "TTL": TTLEstimation,
+        "VVL": VVLEstimation,
+        "ZL": ZLEstimation,
+        "W": WEstimation,
+        "VVJ": VVJEstimation,
+        "TTJ": TTJEstimation,
+        "ZJ": ZJEstimation,
+        "VH125": VHEstimation,
+        "WH125": WHEstimation,
+        "ZH125": ZHEstimation,
+        "ttH125": ttHEstimation,
+        "ggHWW125": ggHWWEstimation,
+        "qqHWW125": qqHWWEstimation,
+    }
+
+    pnameToEstD.update(
+        {ggH_htxs: lambda era, directory, channel,
+         friend_directory:
+         ggHEstimation(
+             ggH_htxs, era, directory, channel,
+             friend_directory=friend_directory)
+         for ggH_htxs in ggHEstimation.htxs_dict})
+    pnameToEstD.update(
+        {qqH_htxs: lambda era, directory, channel,
+         friend_directory:
+         qqHEstimation(
+             qqH_htxs, era, directory, channel,
+             friend_directory=friend_directory)
+         for qqH_htxs in qqHEstimation.htxs_dict})
+    # define sets to select the correct processes
+    trueTauBkgS = {"ZTT", "TTT", "VVT"}
+    leptonTauBkgS = {"ZL", "TTL", "VVL"}
+    jetFakeBkgS = {"ZJ", "W", "TTJ", "VVJ"}
+    jetFakeBkgD = {
+        "et": jetFakeBkgS,
+        "mt": jetFakeBkgS,
+        "tt": jetFakeBkgS,
+        "em": {"W"},
+    }
+
+    # collect all MC backgrounds
+    MCBkgDS = {
+        cn_: trueTauBkgS | leptonTauBkgS | jetFakeBkgD[cn_]
+        for cn_ in selectedChannels}
+
+    # defines the signal sets
+    ww_nicks = set()  # {"ggHWW125", "qqHWW125"}
+    if args.gof_variable is None:
+        signal_nicks = {
+            "WH125", "ZH125", "VH125", "ttH125"} | {
+            ggH_htxs for ggH_htxs in ggHEstimation.htxs_dict} | {
+            qqH_htxs for qqH_htxs in qqHEstimation.htxs_dict} | ww_nicks
+    else:
+        signal_nicks = {"ggHWW125", "qqHWW125"} | ww_nicks
+
+    # Generate dict mapping processnames to proceses
+    processes = {}
+    for chname_, ch_ in selectedChannelsTuples:
+        pL_ = {"data_obs"} | signal_nicks | MCBkgDS[chname_] | {"EMB"}
```

Comment:
Double check this

Reply:
doesnt the QCDEstimation_SStoOS_MTETEM need embedding?